### ISSUE TITLE
Operational issues with await

### ISSUE THREAD

**Issue by one-null-pointer** [SoheilKhodayari/TheThing#3]
Hello author, sorry to disturb you, I am a student and interested in your dom-clobbering project, and have been learning and trying to use your theting recently, but there is a problem running the Dynamic Analyzer of the third module, and after running it says that line 493 there is "await is only valid in async functions and the top level bodies of."  modules"I checked the relevant information to show whether the await keyword is placed in the function declared by async, but it seems that there is no asyns in the 484 lines in the "force-execution" code, can you tell me which step or compiler version is the problem, thank you very much
![issue](https://user-images.githubusercontent.com/100776421/224027038-c3e3872f-8bc5-4958-98db-ebafdec343b2.png)

**Comment by SoheilKhodayari** [comment]
Hi, thanks for your report. I pushed a commit to fix the issue.

**Comment by one-null-pointer** [comment]
Hello, author. It's an honor to receive your reply and solution, and I'm sorry for the delay in my response for a few days. My server crashed, so I was delayed for a few days. Firstly, I couldn't get the latest hash value version of JAW for some unknown reasons, but since you only made changes to the "force_execution.js" file in the project, I just overwrote the content of this file. This change indeed solved the running problem related to "await", but new problems also emerged. Specifically, it prompts an error message in line 46 of the "domc_payload generator.js" file, showing "Cannot read properties of undefined (reading 'line')". The Static Analyzer should have data after running, and there is no error during the running process. I guess this Dynamic Analyzer error is because the object has no value. Is this due to the variable changes caused by the update of "force_execution.js", or is it a problem with my data reading?
![jaw](https://user-images.githubusercontent.com/100776421/225265006-4c39b35d-f2bc-43fc-b28c-006e87436516.png)
![line](https://user-images.githubusercontent.com/100776421/225265055-acb8003f-2184-48cd-afd1-4fb89a4e1efa.png)

**Comment by SoheilKhodayari** [comment]
Hi, thanks for your report, this issue was due to an inconsistency in line formats that emerged during my previous update. I pushed a fix for this by making two changes, please see this [commit](https://github.com/SoheilKhodayari/JAW/commit/d166c4149f246e4496b6ce0f0be49060af264a47).

Regarding your problem with `git`, you should be able to simply do:

```
$ cd JAW
$ git pull
```

**Comment by one-null-pointer** [comment]
Hello author, all the previous problems have been solved with your help, but I have a new problem in the run, the running prompt "Error: Protocol error (Debugger.getScriptSource): No script for id: ", is this because of my own debugger plugin problem? I can't find a solution to the problem on the Internet, how do I solve this problem?
![scripterror](https://user-images.githubusercontent.com/100776421/226250052-edaf1c9f-d488-42ac-aef6-824fea015964.png)

**Comment by SoheilKhodayari** [comment]
@one-null-pointer Thanks for your report. I could not reproduce this problem, could you please post here the exact URL for which the error occured?

PS: this issue seem to affect [Chrome DevTools Protocol](https://chromedevtools.github.io/devtools-protocol/tot/Debugger/) and [puppeteer](https://github.com/puppeteer/puppeteer), where a script id returned by the [`Debugger.scriptParsed`](https://chromedevtools.github.io/devtools-protocol/tot/Debugger/#event-scriptParsed) method of the CDP is not found by the [`Debugger.getScriptSource`](https://chromedevtools.github.io/devtools-protocol/tot/Debugger/#method-getScriptSource) method, causing the CDP connection to drop.  From the picture you posted, this seem to happen when the code is getting the scripts' content with `Debugger.getScriptSource` for each parsed script event.

**Comment by SoheilKhodayari** [comment]
To get around this issue, I added error handing for [`Debugger.getScriptSource`](https://chromedevtools.github.io/devtools-protocol/tot/Debugger/#method-getScriptSource). This should avoid breakage. 

For a proper fix, it has to be investigated why, for some random webpages, an id returned by  [`Debugger.scriptParsed`](https://chromedevtools.github.io/devtools-protocol/tot/Debugger/#event-scriptParsed) method of chrome DevTools is not accepted by `Debugger.getScriptSource` method (which should be).

**Comment by one-null-pointer** [comment]
Hello, sorry to bother you again. The good news is that with your help, the code can run basically. However, I have a few questions and feedback on testing these days.
Question 1:
First, I tried to run Static Analyzer on google.com and encountered a new error message, which said "[Error] problem in mangling TypeError: Cannot read properties of undefined (reading 'length')". Is this because it is not defined?
![1](https://user-images.githubusercontent.com/100776421/228752789-2b744072-6110-4571-b74b-3d3099393fbd.png)
Question 2:
If I understand the code correctly, the final result of a DOM attack is reported in the corresponding webpage's "clobbering.out.json" file, right? Because when I ran the latest code on several websites, the results were all empty, which made me unsure.
![2](https://user-images.githubusercontent.com/100776421/228752821-d01b15f9-9717-4736-9145-e1c740a0e4c5.png)
Question 3:
In addition, these days I have been trying to explore solutions to the problem of obtaining scripts, but I couldn't find any solutions. Worth mentioning, when I repeatedly tested a certain website, the id did not always remain the same. For example, for http://jy.gzhu.edu.cn/unijob/index.php/web/Index/jobs-brief-detail?id=VTUXSSK, sometimes it would appear as 13/8/7, with the probability of id 7 being the highest. Also, during my repeated operations under the Dynamic Analyzer module, I was able to analyze it successfully once, but unfortunately the result was empty. This means that my previous statement about the suspicious source of destruction being the inability to capture the id is incorrect. I analyzed the code and wanted to confirm whether all URLs will call Debugger.scriptParsed and Debugger.getScriptSource for analysis, and the inability to capture the corresponding id script for some URLs will not affect the analysis of other subnets.
![3](https://user-images.githubusercontent.com/100776421/228752873-cfd07dbc-2fb8-4a25-929f-00d5ed0c25a6.png)
Question 4:
When you tested the three website examples I provided earlier, did you also encounter situations where the id script could not be obtained? If not, is it because I am in China and unable to access the corresponding script, resulting in the inability to analyze the DOM destructiveness of the website?
